src/leap/mx/mail_receiver.py

```diff
@@ -145,7 +146,7 @@ def _start_watching_dir(self, dirname, recursive):
         directory = filepath.FilePath(dirname)
         try:
             if not directory.isdir():
-                raise OSError("Not a directory: '%s'" % directory.path)
+                mkdir_p(dirname)
```

Comment:
The permissions and ownership:group should be set here, owner:group (right now leap-mx is being run as root, but we don't want leap-mx to be running as root forever). Also, shouldn't this check to see that the directory creation actually happened?